install/error-handling.sh

```diff
@@ -26,16 +38,44 @@ send_event() {
   # If we haven't sent the envelope file, make it and send to Sentry
   # The format is documented at https://develop.sentry.dev/sdk/envelopes/
   # Grab length of log file, needed for the envelope header to send an attachment
-  local file_length=$(wc -c <"$basedir/$log_file" | awk '{print $1}')
+  local file_length=$(wc -c <$log_path | awk '{print $1}')
+
   # Add header for initial envelope information
-  echo '{"event_id":"'$event_hash'","dsn":"'$SENTRY_DSN'"}' >$envelope_file_path
+  jq -n -c --arg event_id "$event_hash" \
+    --arg dsn "$SENTRY_DSN" \
+    '$ARGS.named' >$envelope_file_path
   # Add header to specify the event type of envelope to be sent
   echo '{"type":"event"}' >>$envelope_file_path
-  # Add traceback message to event
-  echo '{"message":"'$traceback_escaped'","level":"error"}' >>$envelope_file_path
+
+  # Next we construct the meat of the event payload, which we build up
+  # inside out using jq
+  # See https://develop.sentry.dev/sdk/event-payloads/
+  # for details about the event payload
+
+  # First, create the breadcrumb payload
+  # https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/
+  breadcrumbs=$(generate_breadcrumb_json | jq -s -c)
+  # Next we need the exception payload
+  # https://develop.sentry.dev/sdk/event-payloads/exception/
+  # but first we need to make the stacktrace which goes in the exception payload
+  frames=$(echo "$traceback_json" | jq -s -c)
+  stacktrace=$(jq -n -c --argjson frames $frames '$ARGS.named')
+  exception=$(
+    jq -n -c --arg "type" Error \
```

Comment:
guessing here is where we could use the trap code? not sure if that's super helpful though because it would be either ERR or TERM

Reply:
Yeah I don't think using the trap code would be useful, maybe I should just leave as is?

Reply:
I'm fine with that